rdp-server/channels.c

```diff
@@ -671,6 +671,13 @@ static BOOL ogon_processFrontendChannelData(registered_virtual_channel *channel,
 		flags |= CHANNEL_FLAG_SHOW_PROTOCOL;
 	}
 
+	/**
+	 * dynamic channels are always in show protocol mode
+	 */
+	if (channel->channel_type == RDP_PEER_CHANNEL_TYPE_DVC) {
```

Comment:
This won't do anything. At this point we will only ever encounter static channels.
All dynamic channels are encapsulated in the static channel "drdynvc".

Reply:
@nfedera hum yeah you're right the changes should be done in `wts_read_drdynvc_data_first` and `wts_read_drdynvc_data` and always add the header.